File: Data Structures & Algorithms/longest-palindromic-substring/submission-0.js
class Solution {
    /**
     * @param {string} s
     * @return {string}
     */
    longestPalindrome(s) {

        let result=""

        function expandAroudnCenter(left,right){

            while(left>=0&&right<s.length&&s[left]===s[right]){
                left--;
                right++
            }

            return s.slice(left+1,right)
            
        }

        for(let i=0 ;i <s.length;i++)
        {
             let oddchar= expandAroudnCenter(i,i);
             let evenchar= expandAroudnCenter(i,i+1)

             if(oddchar.length>result.length){
                result=oddchar
             }
              if(evenchar.length>result.length){
                result=evenchar
             }
        }

        return result
    }
}

File: Data Structures & Algorithms/sliding-window-maximum/submission-0.java
class Solution {
    public int[] maxSlidingWindow(int[] nums, int k) {
         
       int left =0 , right= 0;
     
       Deque<Integer>q= new LinkedList<>();
       int [] result= new int[ nums.length-k+1];

       while(right<nums.length)
       {
          while(!q.isEmpty() && nums[q.getLast()]<nums[right])
          {
            q.removeLast();
          }
          q.add(right);
          if(left>q.getFirst())
          {
            q.removeFirst();
          }
          if(right+1>=k)
          {
            result[left]=nums[q.peekFirst()];
            left++;
          }
          right++;
       }
        
        return result;
    }
}
